x-pack/plugins/ml/public/data_frame/components/define_pivot/define_pivot_summary.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { SFC, useContext } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiComboBoxOptionProps,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiText,
+} from '@elastic/eui';
+
+import { AggListSummary } from '../../components/aggregation_list';
+import { GroupByList } from '../../components/group_by_list/list';
+import { PivotPreview } from './pivot_preview';
+
+import { Dictionary } from '../../../../common/types/common';
+import {
+  DropDownLabel,
+  DropDownOption,
+  IndexPatternContext,
+  Label,
+  OptionsDataElement,
+  pivotSupportedAggs,
+  SimpleQuery,
+} from '../../common';
+
+const defaultSearch = '*';
+const emptySearch = '';
+
+interface Props {
+  search: string;
+  groupBy: Label[];
+  aggList: Label[];
+}
+
+export const DefinePivotSummary: SFC<Props> = ({ search, groupBy, aggList }) => {
+  const indexPattern = useContext(IndexPatternContext);
+
+  if (indexPattern === null) {
+    return null;
+  }
+
+  const fields = indexPattern.fields
+    .filter(field => field.aggregatable === true)
+    .map(field => ({ name: field.name, type: field.type }));
```

Comment:
I didn't create an interface since I think TS is successfully inferring the type and passing it on. For example, if I'd later on in the code try to do something like `field.name2 = "asdf";` TS will complain that `name2` is a proper attribute of that object.